File: apps/web/lib/auth-mode.ts
import { webConfig } from "./config";

export type FormaAuthMode = "local" | "clerk";

export function formaAuthMode(): FormaAuthMode {
  const value = webConfig.authMode;
  if (!value) {
    throw new Error('FORMA_AUTH_MODE is required. Expected "local" or "clerk".');
  }
  if (value !== "local" && value !== "clerk") {
    throw new Error(`Invalid FORMA_AUTH_MODE=${JSON.stringify(value)}. Expected "local" or "clerk".`);
  }
  return value;
}

export function clerkAuthRequired() {
  return formaAuthMode() === "clerk";
}

File: apps/api/auth_mode.py
"""Explicit authentication mode for the Forma backend."""

from __future__ import annotations

import logging
from blueprint_core.config import config
from typing import Literal


logger = logging.getLogger(__name__)

AuthMode = Literal["local", "clerk"]
AUTH_MODE_ENV = "BLUEPRINT_AUTH_MODE"
AUTH_MODES = {"local", "clerk"}


def blueprint_auth_mode() -> AuthMode:
    """Return the explicitly configured authentication mode."""
    value = (config.get(AUTH_MODE_ENV) or "").strip().lower()
    if not value:
        message = f"{AUTH_MODE_ENV} is required. Expected 'local' or 'clerk'."
        logger.critical(message)
        raise RuntimeError(message)
    if value not in AUTH_MODES:
        message = (
            f"Invalid {AUTH_MODE_ENV}={value!r}. "
            "Expected 'local' or 'clerk'."
        )
        logger.critical(message)
        raise RuntimeError(message)
    return value  # type: ignore[return-value]


def clerk_auth_required() -> bool:
    return blueprint_auth_mode() == "clerk"


__all__ = ["AUTH_MODE_ENV", "AuthMode", "blueprint_auth_mode", "clerk_auth_required"]
